File: QTD2.Domain/Entities/Core/DifSurvey/DIFSurvey_Employee.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QTD2.Domain.Entities.Core
{
    public class DIFSurvey_Employee : Common.Entity
    {
        public int DIFSurveyId { get; set; }
        public int EmployeeId { get; set; }
        public bool? Started { get; set; }
        public bool? Complete { get; set; }
        public int StatusId { get; set; }
        public DateTime? ReleaseDate { get; set; }
        public DateTime? CompletedDate { get; set; }
        public string? Comments { get; set; }
        public virtual DIFSurvey DIFSurvey { get; set; }
        public virtual Employee Employee { get; set; }

        public virtual DIFSurvey_Employee_Status Status { get; set; }
        public virtual ICollection<DIFSurvey_Employee_Response> Responses { get; set; } = new List<DIFSurvey_Employee_Response>();
        public void Update()
        {
            Started = true;
            StatusId = 2;
        }

        public void Completed()
        {
            Started = true;
            CompletedDate = DateTime.Now;
            Complete = true;
            StatusId = 3;
        }
        public DIFSurvey_Employee(DIFSurvey difSurvey,Employee employee,int statusId)
        {
            DIFSurveyId = difSurvey.Id;
            EmployeeId = employee.Id;
            DIFSurvey = difSurvey;
            Employee = employee;
            StatusId = statusId;
        }

        public DIFSurvey_Employee() { }
    }
}

File: QTD.SPA/src/app/_DtoModels/DIFSurvey/DIFSurvey_Employee.ts
import { Entity } from "@models/Entity";
import { DIFSurvey } from "./DIFSurvey";
import { Employee } from "@models/Employee/Employee";
import { DIFSurvey_Employee_Status } from "./DIFSurvey_Employee_Status";
import { DIFSurvey_Employee_Response } from "./DIFSurvey_Employee_Response";

export class DIFSurvey_Employee extends Entity   {
    difSurveyId!:string;
    employeeId!:string;
    started:boolean;
    complete:boolean;
    statusId!:string;   
    releaseDate: Date | null;
    completedDate: Date | null;
    comments:string;
    responses:DIFSurvey_Employee_Response[];
    status:DIFSurvey_Employee_Status;
    employee:Employee;
    difSurvey:DIFSurvey;
}